.github/workflows/bundleTest-thor.yml

```diff
@@ -0,0 +1,238 @@
+          while read bundle
+          do
+              bundleRunPath=${bundle%/ecl}                         # remove '/ecl' from the end of the $bundle
+              bundlePath=${bundleRunPath%/OBTTests};       # remove '/OBTTests' from the end of the $bundleRunPath if exists
+              bundleName=${bundlePath%/test}                    # remove '/test' from the end of the $bundlePath if exists
+              bundleName=$(basename $bundleName )         # remove path from $bundleName
+              
+              if [[ "$bundle" =~ "LearningTrees" ]]
+              then
+                  # add a warning supression parameter in the file
+                  for file in "RegressionTestModified.ecl" "ClassificationTestModified.ecl"
+                  do
+                      if [[ $( egrep -c '#ONWARNING\(30004' $bundle/$file ) -eq 0 ]]
+                      then
+                          pushd $bundle 
+                          cp -fv $file $file-back
+                          # Insert a comment and the "#ONWARNING" after the Copyright header
+                          sed -i '/## \*\//a \\n// Patched by the bundleTest on '"$( date '+%Y.%m.%d %H:%M:%S')"' \n#ONWARNING(30004, ignore); // Do not report execute time skew warning' $file
+                          popd
+                      fi
+                  done
+              fi
+              
+              if [[ ! "${BUNDLES_TO_TEST[*]}" =~ "$bundleName"  ]]
+              then
+                  continue
+              fi
+              
+              pushd $bundleRunPath
+          
+              /opt/HPCCSystems/testing/regress/ecl-test run -t thor --config /opt/HPCCSystems/testing/regress/ecl-test.json --timeout 3600 -fthorConnectTimeout=3600 --pq 1 --ef ClassicTestModified.ecl,SVCTest.ecl,ClassificationTestModified.ecl
+              
+              retCode=$( echo $? )
+          
+              if [ ${retCode} -eq 0 ] 
+              then
+                  ProcessLog "$bundleName" "thor"
+              fi
+            
+              popd
+  
+          done< <(find . -iname 'ecl' -type d | sort )
+      - name: Generate ZAP files
+        if: ${{ ! inputs.Generate-Zap == '' }} 
+        run: |  
+          IFS=' ' read -a ML_GENERATE_ZAP_FOR <<< ${{ inputs.Generate-Zap }}
+          if [  ${#ML_GENERATE_ZAP_FOR[@]} -ne 0 ]
+          then
+              for test in ${ML_GENERATE_ZAP_FOR[*]}
+              do 
+                  wuid=$(ecl getwuid -n $test --limit 1)
+                  if [ -n $wuid ]
+                  then
+                      ecl zapgen $wuid  --path /home/runner/HPCCSystems-regression/zap --inc-thor-slave-logs
+                  fi
+              done
+          fi
+
+      - name: Check for Core files
+        run: |
+          NUM_OF_ML_CORES=( $(sudo find /var/lib/HPCCSystems/ -iname 'core*' -mtime -1 -type f -exec printf "%s\n" '{}' \; ) )
```

Comment:
Same is true for Core handling as was in ZAP file generation. One more extra thing, core file can created without any test failure (hidden crash). So, if any core file found, the trace file should be a part of the artifact and need to upload.